### PR TITLE
feat: implement events

### PR DESCRIPTION
**Pull request by DjDeveloperr**
towards #1 

* buffered reading. very large packets were broken.
* refactored into low level `DiscordIPC` and high level `Client`
* implement events based on `AsyncIterator`

### CODE REVIEW

**Comment by DjDeveloperr** [comment]
I was thinking about that we can probably keep `DiscordIPC` for sole purpose of interacting with Discord IPC at low level, and add an `Client` which abstracts away things such as commands, events.

**Comment by DjDeveloperr** [comment]
Also constructing class the way it's done right now adds some unnecessary complexities I feel like and we're locked to use unique symbols. Why not let the constructor accept `Deno.Conn` and `createClient` or `DiscordIPC.connect` static method for finding Conn and constructing it? It could also help us write tests by mocking the IPC.

File: src/conn.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import { encode, findIPC } from "./util.ts";
+
+const _ipcHandle = Symbol("[[ipc]]");
+const _header = Symbol("[[header]]");
+const _headerView = Symbol("[[headerView]]");
+const _writers = Symbol("[[writers]]");
+const _emit = Symbol("[[emit]]");
+const _eventLoop = Symbol("[[eventLoop]]");
+const _read = Symbol("[[read]]");
+const _breakEventLoop = Symbol("[[breakEventLoop]]");
```
[the remark this answers is not in the document]
Comment (DjDeveloperr) [reply src/conn.ts]: Done

**Comment by DjDeveloperr** [comment]
Also added `Client` implementation to this PR

File: examples/messages.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Client, Message } from "../mod.ts";
+
+const client = new Client({
+  id: Deno.env.get("CLIENT_ID")!,
+  secret: Deno.env.get("CLIENT_SECRET")!,
+  scopes: ["rpc", "messages.read"],
+});
+
+(async () => {
+  for await (const event of client) {
```
Comment (littledivy) [inline examples/messages.ts:10]: wrapping inside async is not required with TLA

Comment (DjDeveloperr) [reply examples/messages.ts:10]: that for await loop will block forever until client is disconnected

**Review by littledivy — APPROVED** [review]
LGTM Thank you!

**Pull request merged**